tensorly/tenalg/proximal.py

```diff
@@ -248,3 +248,196 @@ def hals_nnls(UtM, UtU, V=None, n_iter_max=500, tol=10e-8,
+        x = tl.zeros(tl.shape(UtM))
+
+    # Parameters
+    momentum_old = tl.tensor(1.0)
+    norm_0 = 0.0
+    x_update = tl.copy(x)
+
+    for iteration in range(n_iter_max):
+        if isinstance(UtU, list):
+            x_gradient = - UtM + tl.tenalg.multi_mode_dot(x_update, UtU, transpose=False) + sparsity_coef
+        else:
+            x_gradient = - UtM + tl.dot(UtU, x_update) + sparsity_coef
+
+        if non_negative is True:
+            x_gradient = tl.where(lr * x_gradient < x_update, x_gradient, x_update/lr)
+
+        x_new = x_update - lr * x_gradient
+        momentum = (1 + tl.sqrt(1 + 4 * momentum_old ** 2)) / 2
+        x_update = x_new + ((momentum_old - 1) / momentum) * (x_new - x)
+        momentum_old = momentum
+        x = tl.copy(x_new)
+        norm = tl.norm(lr * x_gradient)
+        if iteration == 1:
+            norm_0 = norm
+        if norm < tol * norm_0:
+            break
+    return x
+
+
+def active_set_nnls(Utm, UtU, x=None, n_iter_max=100, tol=10e-8):
+    """
+     Active set algorithm for non-negative least square solution.
+
+     Computes an approximate non-negative solution for Ux=m linear system.
+
+     Parameters
+     ----------
+     Utm : vectorized ndarray
+        Pre-computed product of the transposed of U and m
+     UtU : ndarray
+        Pre-computed Kronecker product of the transposed of U and U
+     x : init
+        Default: None
+     n_iter_max : int
+         Maximum number of iteration
+         Default: 100
+     tol : float
+         Early stopping criterion
+
+     Returns
+     -------
+     x : ndarray
+
+     Notes
+     -----
+     This function solves following problem:
+     .. math::
+        \\begin{equation}
+             \\min_{x} ||Ux - m||^2
+        \\end{equation}
+
+     According to [1], non-negativity-constrained least square estimation problem becomes:
+     .. math::
+        \\begin{equation}
+             x' = (Utm) - (UTU)\\times x
+        \\end{equation}
+
+     Reference
+     ----------
+     [1] : Bro, R., & De Jong, S. (1997). A fast non‐negativity‐constrained
+           least squares algorithm. Journal of Chemometrics: A Journal of
+           the Chemometrics Society, 11(5), 393-401.
+     """
+    
+    if x is None:
+        x_vec = tl.zeros(tl.shape(UtU)[1])
+    else:
+        x_vec = tl.base.tensor_to_vec(x)
+
+    x_gradient = Utm - tl.dot(UtU, x_vec)
+    passive_set = x_vec > 0
+    active_set = x_vec <= 0
+    s = tl.zeros(tl.shape(x_vec), **tl.context(x_vec))
+
+    for iteration in range(n_iter_max):
+
+        if iteration > 0 or tl.all(x_vec == 0):
+            indice = tl.argmax(x_gradient)
+            passive_set = tl.index_update(passive_set, tl.index[indice], True)
+            active_set = tl.index_update(active_set, tl.index[indice], False)
+        # To avoid singularity error when initial x exists
+        try:
+            a = tl.tensor(tl.to_numpy(UtU)[passive_set, :][:, passive_set], **tl.context(UtU))
```

Comment:
I've been debating about this and trying to make it work - the main issue is TensorFlow: I am thinking to either not support tensorflow here or checking for the backend and having a separate case for it. What do you think @caglayantuna @cohenjer 

More generally I'm more and more inclined to phase out tensorflow support (we have jax anyway).

Reply:
Yes I agree, it is difficult to handle tensorflow for some situations. I changed the codes as you suggested. As a suggestion, seperate case could be directing active set to fista with an informative warning if backend is tensorflow.

Reply:
It does seem like the Tensorflow backend is causing us a lot of problems. Not sure if people are actually using it? Maybe a quick survey on this backend usage would help checking if it is worth the effort to support it at the moment?

Reply:
Yes, I'll create a survey. @caglayantuna that sounds like a good solution at least for now - we can just make it work for backends other than tensorflow and at the beginning of the nn_tucker, if method is 'fista' then check for the backend and return an error if is tensorflow.